### PR TITLE
MEL-441: Update react scripts

### PR DESCRIPTION
**Pull request by edmofro**
### Issue #:
Towards MEL-441

### Changes:
Update react scripts to 4.0.3, which is the latest. This probably has a bunch of improvements in it, but specifically interested in:
- Uses a more recent version of `terser`, which has a fix in it for using too many cpus and running out of memory
- Forces us to fix linter errors, most significantly the import cycles, which may have been a significant cause of the build being resource intensive

### CODE REVIEW

File: packages/psss/src/components/Table/CellComponents/ArchivedAlertMenuCell.js
Comment (kael89) [inline packages/psss/src/components/Table/CellComponents/ArchivedAlertMenuCell.js:6]: I haven't seen this pattern again (passing the context as prop), instinctively it seems a bit unexpected but it may be fine - would like to get another opinion from @tcaiger 

I assume that the motivation for this change was to avoid an import dependency cycle, and I know that this is sometimes tricky to achieve, so I'm personally not fussed either way!

File: packages/lesmis/src/constants/page.js
Comment (kael89) [inline packages/lesmis/src/constants/page.js:115]: I'd be happy for us to disable this ESLint rule ([react/no-unescaped-entities](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-unescaped-entities.md))

File: .eslintrc
Comment (kael89) [inline .eslintrc:28]: I generally like this rule , since it makes it very clear to see whether something comes from `state` or from `props`. However, I understand that there are cases where it may be more elegant to not destructure and just use eg. `this.props.aFunction()`, so disabling it in the top level seems like a good call.

File: packages/admin-panel/src/widgets/InputField.js
Comment (kael89) [inline packages/admin-panel/src/widgets/InputField.js:50]: I like this refactoring!

File: packages/admin-panel/src/widgets/InputField/registerInputFields.js
Comment (kael89) [inline packages/admin-panel/src/widgets/InputField/registerInputFields.js:15]: Clever pattern!

File: packages/admin-panel/src/widgets/InputField/Autocomplete.js
Comment (kael89) [inline packages/admin-panel/src/widgets/InputField/Autocomplete.js:42]: This is already registered in `registerInputFields`:

```suggestion
```

**Review by kael89 — APPROVED** [review]
Great refactor! So the new version of `react-scripts` enforces us to fix linting errors? That's interesting! I'd assume that they would allow users to skip this, but I definitely love the cleanup in this PR ♥

It makes me think whether it would pay off to be a bit stricter with following the linting rules. I think that as soon as you get used to them, it's very easy to follow them in practice. Plus, if a rule is too strict we can turn it into a warning, or disable it.

A few minor suggestions but pre-approving

File: packages/psss/src/components/Table/CellComponents/ArchivedAlertMenuCell.js
Comment (kael89) [reply packages/psss/src/components/Table/CellComponents/ArchivedAlertMenuCell.js:6]: Or could we extract `ArchiveTableContext` in a new file and thus avoid the dependency loop?

**Comment by kael89** [comment]
@edmofro one last comment I forgot: there is a file in this PR that doesn't seem to be used (`packages/admin-panel/src/widgets/InputField/Enum.js`)

File: packages/admin-panel/src/widgets/InputField/Autocomplete.js
Comment (edmofro) [reply packages/admin-panel/src/widgets/InputField/Autocomplete.js:42]: Ooops, left over from a first attempt, thanks for catching!

File: packages/psss/src/components/Table/CellComponents/ArchivedAlertMenuCell.js
Comment (edmofro) [reply packages/psss/src/components/Table/CellComponents/ArchivedAlertMenuCell.js:6]: @chris-bes could you follow up on this one?

Comment (tcaiger) [reply packages/psss/src/components/Table/CellComponents/ArchivedAlertMenuCell.js:6]: Yes, I agree with Kostas. thanks

**Comment by edmofro** [comment]
Thanks for the review @kael89! I'm sure there is a way to ask react-scripts to be less strict, but it was a great opportunity to keep in place something that will give us a bit more discipline, especially around circular dependencies which I strongly suspect have been contributing to our heavy resource use during builds

All my bits fixed up, @chris-bes will respond after lunch then this will be ready for re-review

**Comment by kael89** [comment]
> I'm sure there is a way to ask react-scripts to be less strict, but it was a great opportunity to keep in place something that will give us a bit more discipline, especially around circular dependencies which I strongly suspect have been contributing to our heavy resource use during builds

Definitely a good call, kudos for that!

File: packages/psss/src/components/Table/CellComponents/ArchivedAlertMenuCell.js
Comment (chris-bes) [reply packages/psss/src/components/Table/CellComponents/ArchivedAlertMenuCell.js]: Definitely was a poor effort to fix the dependency cycle issue 🤦‍♂️ . Thanks for pointing that out @kael89 !

**Review by kael89 — APPROVED** [review]
Nice cleanup!!

**Comment by chris-bes** [comment]
Have also addressed my bits

File: packages/web-frontend/src/components/Marker/markerIcons.js
Comment (tcaiger) [reply packages/web-frontend/src/components/Marker/markerIcons.js:27]: Hey @chris-bes, I just saw this conversation. I have another PR open to move all the marker code to ui-components along with all the map-components from web-frontend. I ran into a similar issue there and resolved it in the rollup build config. So let's either wait until those changes are merged in to dev or come up with a quick work around so that you can move on. I'll send you a PM.

Comment (tcaiger) [reply packages/web-frontend/src/components/Marker/markerIcons.js:27]: @edmofro What do we think about hardcoding the disaster icons to tupaia orange as a quick fix for this temporary code? (As far as I can tell they are always displayed orange). Then when my map refactoring is complete this code will be deleted.

Comment (edmofro) [reply packages/web-frontend/src/components/Marker/markerIcons.js:27]: Eeek yes sorry @tcaiger could you take this over now @chris-bes is off on Tamanu? Hardcoding to tupaia orange is totally fine with me.

Comment (tcaiger) [reply packages/web-frontend/src/components/Marker/markerIcons.js:27]: @edmofro - yes can do. My understanding is that this file is the last outstanding issue. Is that correct?

Comment (edmofro) [reply packages/web-frontend/src/components/Marker/markerIcons.js:27]: Yes plus a merge from dev with some conflicts to resolve, and my accidental commit before lunch (was meant for a different branch). Thanks heaps for taking care of the svg stuff, I'll rebase the branch now to get rid of that commit and resolve the conflicts!

Comment (edmofro) [reply packages/web-frontend/src/components/Marker/markerIcons.js:27]: Rebased and resolved conflicts. Hopefully I got it all right. Ready for your input on the svgs @tcaiger!

**Comment by tcaiger** [comment]
@edmofro and @kujo4pm - Ready for review. I just added 1 new commit.

File: packages/web-frontend/package.json
Comment (tcaiger) [inline packages/web-frontend/package.json:98]: F some reason those settings cause an error with svg imports. Using the defaults works fine though and the list of browsers it includes is very similar so I am keen to go for the defaults. See here for a list of the browsers included in each. https://beyondessential.slack.com/archives/C01UPMP1LPN/p1630627300000900

**Comment by edmofro** [comment]
I can't review my own PR unfortunately, but your commit lgtm! Thanks heaps for that @tcaiger 

(@kujo4pm I believe you can ignore that tag ;-) )

**Review by kael89 — APPROVED** [review]
Lgtm too, thanks @tcaiger !

File: packages/web-frontend/package.json
Comment (IgorNadj) [reply packages/web-frontend/package.json:98]: Thanks for that Tom! BTW I ran into the same issue in admin-panel, could you make the same change there as well? Thanks

**Pull request merged**
